python/cuspatial/cuspatial/core/spatial/join.py

```diff
@@ -214,14 +214,15 @@ def quadtree_point_in_polygon(
         raise ValueError(
             "`polygons` Geoseries must contains only polygons geometries."
         )
-
-    points_x = as_column(points.points.x)
-    points_y = as_column(points.points.y)
-
-    poly_offsets = as_column(polygons.polygons.part_offset)
-    ring_offsets = as_column(polygons.polygons.ring_offset)
-    poly_points_x = as_column(polygons.polygons.x)
-    poly_points_y = as_column(polygons.polygons.y)
+    points_data = points.points
+    points_x = as_column(points_data.x)
+    points_y = as_column(points_data.y)
```

Comment:
I think we should allow users to avoid packing points into a GeoSeries if desired.

Since GeoSeries is a `Union<List<List<List<Float32>>>`, each 8-byte point pair requires an additional 13 bytes (`1 + 4 * 3` ) of indices, which is prohibitively expensive when the dataset is large.

I suggest we either change the API to accept separate x/y columns again (while still accepting and unpacking the points from a GeoSeries), or provide some sort of wrapper for a pair of x/y columns that can be accessed the same way as the GeoSeries.

At the moment, here's my workaround:

```python
class Points:
    def __init__(self, x, y):
        self.x = x
        self.y = y
        self.points = self
        self._column = self
        self._meta = self
        self.input_types = Series([Feature_Enum.POINT.value])

    def __len__(self) -> int:
        return len(self.x)


x = Series([ ... ])
y = Series([ ... ])
# Wrap x and y cols in a Points instance which duck-types
# GeoSeries' accessors ala `p.points.x` and `p.points.y`
quadtree_on_points(Points(x, y), ...)
```

Reply:
Historically it looks like the API changed in https://github.com/rapidsai/cuspatial/pull/934 where `quadtree_point_in_polygon` was updated to accept a `GeoSeries` points and polygon inputs. I suppose @isVoid is best to make the call on whether we should change the API back to accept x/y points again.

Reply:
See #1486